src/front/glsl_new/ast.rs

```diff
@@ -39,6 +39,19 @@ impl Program {
             },
         }
     }
+
+    pub fn binary_expr(
+        self: &mut Program,
```

Comment:
why is this not just `&mut self`?

Reply:
Because I'm still a rust noob :P

Simply copy pasted the external function and made it compile 🤦‍♂️